### PR TITLE
Performance

### PR DESCRIPTION
**Pull request by badger-99**
## This PR will introduce the following changes.
- An `email` column to the `owners` table.
- Additional data in the `owners` and `visits` tables.
- Performance optimization on queries.

<hr/>

## Performance optimization has been done for the following queries.

 - `SELECT COUNT(*) FROM visits where animal_id = 4;`
  > **Before Optimization**
Query1-before

  > **After Optimization**
Query1-after

<br/>

 - `SELECT * FROM visits where vet_id = 2;`
  > **Before Optimization**
Query2-before

  > **After Optimization**
Query2-after

<br/>

 - `SELECT * FROM owners where email = 'owner_18327@mail.com';`
  > **Before Optimization**
Query3-before

  > **After Optimization**
Query3-after

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -48,14 +41,24 @@ CREATE TABLE specializations(
...
+ADD COLUMN email VARCHAR(120);
+
+-- Added index to animals_id column on visits table
+CREATE INDEX idx_animal_id ON visits (animals_id ASC);
+
+-- Added index to vets_id column on visits table
+CREATE INDEX visits_covering_idx ON visits(vets_id) INCLUDE (visits_id, animals_id, vets_id, date_of_visit);
+
+--Create index to email column on owners table
+CREATE INDEX owner_email ON owners(email ASC);
```
Comment (topeogunleye) [inline schema.sql:64]: 👏 **Optional Suggestion** 👏

Adding a case-insensitive index on the email column is a thoughtful consideration, especially when dealing with email addresses, which are not case-sensitive by convention. It allows for more flexible and user-friendly queries. To implement this, you can use the `lower()` function in your index creation:

```sql
CREATE INDEX ON owners ((lower(email)));
```

By doing this, you ensure that queries for emails are case-insensitive, making the database more user-friendly. However, please note that this suggestion is optional and might not be directly related to the performance issues you addressed in your project. It's more of an enhancement for user convenience.

Great job on being proactive in thinking about user experience and flexibility! Keep up the good work! 👍🌟

**Review by topeogunleye — APPROVED** [review]
Hi @badger-99, @SanaTameem, @sulaimonaa,


Great job, student! Your project successfully addresses the performance issues by optimizing queries and adding appropriate indexes. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

dwayne-the-rock

## (Highlights) Good Points: 💪 
🚀 **Code Review Summary for Project** 🚀

🕐 **Query Optimization:** You've tackled the three problematic queries, and your "before" and "after" screenshots clearly show significant improvements in execution time.

📊 **Index Implementation:** You've correctly added indexes to the relevant columns, which is a fundamental solution for performance problems. Well done on creating these indexes:

```sql
CREATE INDEX ON visits (animal_id);
CREATE INDEX ON visits (vet_id);
CREATE INDEX ON owners (email);
```

🔍 **Documentation:** Your project documentation includes clear explanations of the issues and the solutions you implemented. You've also followed the optional suggestion to make the email index case-insensitive, which is a thoughtful addition.

Overall, your project demonstrates a solid understanding of query optimization and indexing. Keep up the excellent work, and continue exploring ways to enhance database performance! 👍🌟
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql (hunk 1 above)
Comment (badger-99) [reply schema.sql:64]: Thank you for pointing this issue out. I'm quite new to databases and this is my first project so adding case insensitivity is not something I even thought about.  But now that you have mentioned It, I'll always have it in mind to implement it in future projects.

Comment (topeogunleye) [reply schema.sql:64]: You're very welcome! 😊

It's fantastic to hear that you found the suggestion helpful, especially in the context of your first project. Databases can indeed be a bit tricky at times, but you're on the right track by being open to new ideas and improvements.

Remember, learning and growing as a developer is all about exploring new concepts and continuously enhancing your skills. If you ever have more questions or need guidance in the future, don't hesitate to reach out. We're here to support your journey.

Best of luck with your future projects, and keep up the great work! 👍🚀

Comment (badger-99) [reply schema.sql:64]: Thank you very much for the encouragement, I appreciate it a lot!

**Pull request merged**